test/bin/mdx-test/expect/environment-variable-set/test-case.md

```diff
@@ -21,6 +21,14 @@ And the variable stays available in subsequent blocks.
   - : unit = ()
 ```
 
+Environment variables can contain underscores
+
+```ocaml set-TWO_WORDS=success
```

Comment:
End to end tests are great but we're also trying to improve the unit testing situation in MDX, would you mind making those unit tests in https://github.com/realworldocaml/mdx/blob/master/test/lib/test_label.ml.

You can probably leave one of those two e2e tests, they definitely don't hurt!